src/operator/catch.ts

```diff
@@ -64,6 +65,8 @@ import { subscribeToResult } from '../util/subscribeToResult';
  * @name catch
  * @owner Observable
  */
+export function _catch<T>(this: Observable<T>, selector: (err: any, caught: Observable<T>) => ErrorObservable): Observable<T>;
+export function _catch<T, R>(this: Observable<T>, selector: (err: any, caught: Observable<T>) => ObservableInput<R>): Observable<T | R>;
```

Comment:
When you overload a function _definition_, you need to declare an additional overload implementation as the implementation signature is no longer part of the type.